### PR TITLE
Fix memory.grow bounds and overflow checks for mem64

### PR DESCRIPTION
**Pull request by tlively**
Previously the interpreter only executed overflow and bounds checks for
memory.grow on 32-bit memories. Run the checks on 64-bit memories as well.

### CODE REVIEW

File: src/wasm-interpreter.h [hunk 1]
```diff
@@ -3836,10 +3836,14 @@ class ModuleRunnerBase : public ExpressionRunner<SubType> {
     auto fail = Literal::makeFromInt64(-1, memory->addressType);
     Flow ret = Literal::makeFromInt64(memorySize, addressType);
     uint64_t delta = flow.getSingleValue().getUnsigned();
-    if (delta > uint32_t(-1) / Memory::kPageSize && addressType == Type::i32) {
+    uint64_t maxAddr =
+      addressType == Type::i32 ? uint64_t(uint32_t(-1)) : uint64_t(-1);
```
Comment (sbc100) [inline src/wasm-interpreter.h:3840]: Use `UINT64_MAX` and `UINT32_MAX` here?

File: src/wasm-interpreter.h [hunk 2]
```diff
@@ -3836,10 +3836,14 @@ class ModuleRunnerBase : public ExpressionRunner<SubType> {
     auto fail = Literal::makeFromInt64(-1, memory->addressType);
     Flow ret = Literal::makeFromInt64(memorySize, addressType);
     uint64_t delta = flow.getSingleValue().getUnsigned();
-    if (delta > uint32_t(-1) / Memory::kPageSize && addressType == Type::i32) {
```
Comment (sbc100) [inline src/wasm-interpreter.h:3839]: Strange there wan't event a TODO here!

**Review by kripken — APPROVED** [review]
lgtm % open discussions

File: src/wasm-interpreter.h (hunk 1 above)
Comment (sbc100) [reply src/wasm-interpreter.h:3840]: Yuck, that is so much hard to read than `UINT32_MAX`, no?

Comment (kripken) [reply src/wasm-interpreter.h:3840]: Heh, I don't feel strongly. I guess it is more C++ey and can support all types and all bounds. But I wouldn't object to switching.

Comment (sbc100) [reply src/wasm-interpreter.h:3840]: I get that binaryen is fully in on all the C++ things :)   So lgtm either way, I just wish C++ wasn't so verbose sometimes.

**Pull request merged**